From ccf55d7b7ba8de94a3685bf109eb1cc595decec9 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 251/296] Disable PIN entry after successful quick unlock

Small polish: I still often press the enter button out of muscle memory
when entering my PIN code, which leads to an "invalid PIN" message being
shown for a tiny moment as the screen unlocks. Disable the entry view
after a quick unlock to fix this.
---
 packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java b/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
index c2c50885741..4f2161a0d0c 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardPINView.java
@@ -232,6 +232,7 @@ public class KeyguardPINView extends KeyguardPinBasedInputView {
         if (password != null) {
             if (password.length() > MINIMUM_PASSWORD_LENGTH_BEFORE_REPORT
                     && kpvCheckPassword(password)) {
+                mPasswordEntry.setEnabled(false);
                 mCallback.reportUnlockAttempt(userId, true, 0);
                 mCallback.dismiss(true, userId);
                 resetPasswordText(true, true);
-- 
2.11.0

